### PR TITLE
Issue #7735: Update AbstractChecks to log DetailAST - InterfaceIsType

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #7735: Update AbstractChecks to log DetailAST - InterfaceIsType

diff report : 
https://huganghui.github.io/7735-DetailAST-InterfaceIsType/reports/

### CODE REVIEW

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionInterfaceIsTypeTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+        moduleConfig.addAttribute("allowMarkerInterfaces", "false");
+
+        final String[] expectedViolation = {
+            "3:1: " + getCheckMessage(InterfaceIsTypeCheck.class,
+                    InterfaceIsTypeCheck.MSG_KEY),
+        };
+
+        final List<String> expectedXpathQueries = Arrays.asList(
+                "/INTERFACE_DEF[./IDENT["
```
Comment (baratali) [inline src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionInterfaceIsTypeTest.java]: Indentation seems incorrect (espessially comparing with the same list from testOne). Please recheck formatting

Comment (HuGanghui) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionInterfaceIsTypeTest.java]: Thanks, done, actually, it is testOne whose format is not good and I have changed it.

**Comment by baratali** [comment]
@HuGanghui , I'm ok with the changes. Please provide a diff report.

**Comment by baratali** [comment]
@HuGanghui , ping

**Comment by HuGanghui** [comment]
@baratali I have updated diff report, please review~

**Pull request merged**